File: src/main/java/mods/eln/sim/watchdogs/BipoleVoltageWatchdog.java
package mods.eln.sim.watchdogs;

import mods.eln.sim.electrical.mna.component.Bipole;

public class BipoleVoltageWatchdog extends ValueWatchdog {

    Bipole bipole;

    public BipoleVoltageWatchdog set(Bipole bipole) {
        this.bipole = bipole;
        return this;
    }

    public BipoleVoltageWatchdog setUNominal(double UNominal) {
        this.max = UNominal * 1.3;
        this.min = -max;
        this.timeoutReset = UNominal * 0.10 * 5;

        return this;
    }

    @Override
    double getValue() {
        return bipole.getU();
    }
}

File: src/main/java/mods/eln/sim/mna/BipoleVoltageWatchdog.kt
package mods.eln.sim.mna

import mods.eln.sim.core.ValueWatchdog
import mods.eln.sim.mna.passive.Bipole

class BipoleVoltageWatchdog(internal val bipole: Bipole) : ValueWatchdog() {

    fun setUNominal(UNominal: Double) {
        this.max = UNominal * 1.3
        this.min = -max
        this.timeoutReset = UNominal * 0.10 * 5.0
    }

    override fun getValue(): Double {
        return bipole.getVoltage()
    }
}
